certbot-apache/certbot_apache/augeasparser.py

```diff
@@ -253,6 +289,72 @@ def _aug_get_block_name(self, path):
         name = name.split("[")[0]
         return name
 
+    def _aug_resolve_child_position(self, name, position):
+        """
+        Helper function that iterates through the immediate children and figures
+        out the insertion path for a new AugeasParserNode.
+
+        Augeas also generalizes indices for directives and comments, simply by
+        using "directive" or "comment" respectively as their names.
+
+        This function iterates over the existing children of the AugeasBlockNode,
+        returning their insertion path, resulting Augeas path and if the new node
+        should be inserted before or after the returned insertion path.
+
+        Note: while Apache is case insensitive, Augeas is not, and blocks like
+        Nameofablock and NameOfABlock have different indices.
+
+        :param str name: Name of the AugeasBlockNode to insert, "directive" for
+            AugeasDirectiveNode or "comment" for AugeasCommentNode
+        :param int position: The position to insert the child AugeasParserNode to
+
+        :returns: Tuple of insert path, resulting path and a boolean if the new
+            node should be inserted before it.
+        :rtype: tuple of str, str, bool
+        """
+
+        # Default to appending
+        before = False
+
+        all_children = self.parser.aug.match("{}/*".format(
```

Comment:
@bmw there's a few lines of augeas going in this function that look reasonable but I know I'll miss any weird nuances so would love an extra set of eyes!

Reply:
Other than my comment at https://github.com/certbot/certbot/pull/7497/files#r343358309, the Augeas code in this function looks correct to me!